### PR TITLE
Implement WaitForMetaTxn

### PR DESCRIPTION
**Pull request by Agusx1211**

### CODE REVIEW

File: relayer.go [hunk 1]
```diff
@@ -38,6 +40,11 @@ type Relayer interface {
 
 type MetaTxnID string
 
+var (
```
Comment (pkieltyka) [inline relayer.go:43]: FYI, we have these computed here: https://github.com/0xsequence/go-sequence/blob/master/transactions.go#L94

File: relayer_test.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		To:            callmockContract.Address,
+		Data:          calldata,
+		Value:         big.NewInt(0),
+		GasLimit:      big.NewInt(190000),
+		DelegateCall:  false,
+		RevertOnError: false,
+		Nonce:         nonce,
+	}
+
+	err = signAndSendRawTransaction(t, wallet, stx)
```
Comment (pkieltyka) [inline relayer_test.go:42]: seems we can prob move these methods along with other helpers directly on the *TestChain, we can do it later tho, but feels like the right thing to do.

File: relayer.go [hunk 3]
```diff
@@ -86,6 +93,65 @@ func EncodeTransactionsForRelaying(relayer Relayer, walletConfig WalletConfig, w
...
+	for time.Now().Before(timeoutTime) {
+		block, err := provider.BlockNumber(ctx)
+
+		if err != nil {
+			time.Sleep(time.Second)
+			continue
+		}
+
+		query := ethereum.FilterQuery{
+			FromBlock: big.NewInt(int64(lastBlockNumber) - 12),
```
Comment (PhABC) [inline relayer.go:122]: Think we shouldn't hardcode the 12 value and set it as a variable like in sequence.js.

Comment (Agusx1211) [reply relayer.go:117]: I agree, I will leave a comment so we later add some sort of configuration parameter or something.

**Pull request merged**